test/unittest/unittest_nntrainer_tensor_pool.cpp

```diff
@@ -403,6 +403,83 @@ TEST(TensorPool, validate_memory) {
   EXPECT_NO_THROW(pool.deallocate());
 }
 
+/**
+ * @brief check if data span of two tensor testOverlap
+ *
+ * @param t1 tensor1
+ * @param t2 tensor2
+ */
+static void testNotOverlap(nntrainer::Tensor *t1, nntrainer::Tensor *t2) {
+  char *t1_start = t1->getData<char>();
+  char *t1_end = t1_start + t1->bytes();
+
+  char *t2_start = t2->getData<char>();
+  char *t2_end = t2_start + t2->bytes();
+
+  EXPECT_NE(t1_start, nullptr);
+  EXPECT_NE(t2_start, nullptr);
+  EXPECT_TRUE(!(t1_start < t2_end && t2_start < t1_end))
+    << "t1 and t2 overlaps";
+}
+
+/**
+ * @brief test if t1 is including t2
+ *
+ * @param t1 t1 tensor 1
+ * @param t2 t2 tensor 2
+ */
+[[maybe_unused]] static void testSubset(nntrainer::Tensor *t1,
+                                        nntrainer::Tensor *t2) {
+  char *t1_start = t1->getData<char>();
+  char *t1_end = t1_start + t1->bytes();
+
+  char *t2_start = t2->getData<char>();
+  char *t2_end = t2_start + t2->bytes();
+
+  EXPECT_NE(t1_start, nullptr);
+  EXPECT_NE(t2_start, nullptr);
+  EXPECT_TRUE(t1_start <= t2_start && t2_end < t1_end)
+    << "t2 is not subset of t1";
+}
+
+TEST(TensorPool, create_allocate_has_data_p) {
+  nntrainer::TensorPool pool;
+  nntrainer::Tensor *t1 = nullptr, *t2 = nullptr;
+
+  t1 = pool.create("a", {10}, {0}, nntrainer::TensorLifespan::MAX_LIFESPAN);
```

Comment:
i recommend you to reuse the existing unit tests of `requestTensor` as it is a more extensive set of tests and `create` is an alias for `requestTensor` anyway.

Reply:
This will be automatically done when we remove aliasing of `requestTensor` and `create` in the upcoming PR's.
This is intentionally left untouched to keep PR size small as much as possible as it is already big.